File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/9. Palindrome Number/python-solution/Palindrome-Number.py
class Solution(object):
    def isPalindrome(self, num):
        """
        :type num: int
        :rtype: bool
        """
        # Using the two pointers technique 
        # this solution is better because it does not use another while loop, it just uses one while loop 
        # Time complexity = O(n) , Space complexity = O(1)
        if num < 0:
            return False
        num = str(num) # first change the integer  number to string to access each digit with their idex
        L, R = 0, len(num) - 1
        while L < R:
            if num[L] != num[R]:
                return False
            L += 1
            R -= 1
        return True

File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/9. Palindrome Number/c++-solution/Palindrome-Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
    
        // if (x < 0)
        // return false; // here if we are to change x to str we don't need to check if it is -ve, the while loop will takecare of that
        //uing two pointers technique by changing int to str 
        // Time complexity = O(n) , Space complexity = O(1)
        string num = to_string(x); // in c++ we use to_stirng() method to convert int to str
        int L = 0;
        int R = num.length() - 1;
        while (L < R){
            if (num[L] != num[R])
            return false;
            L++;
            R--;
        }
        return true;
    }
};